src/Neo.CLI/Settings.cs

```diff
@@ -117,4 +120,22 @@ public UnlockWalletSettings(IConfigurationSection section)
             }
         }
     }
+
+    public class NeoNameServiceSettings
+    {
+        public UInt160 Contract { get; }
+
+        public NeoNameServiceSettings(IConfigurationSection section)
+        {
+            if (section.Exists())
+            {
+                if (UInt160.TryParse(section.GetValue(nameof(Contract), string.Empty), out var hash))
+                {
+                    this.Contract = hash;
+                }
+                else
+                    ConsoleHelper.Error("Neo Name Service (NNS): Contract hash is invalid. Check your config.json.");
```

Comment:
throw?

```suggestion
                    throw(new Exception("Neo Name Service (NNS): Contract hash is invalid. Check your config.json."));
```

Reply:
agree

Reply:
why crash the application?

Reply:
Cause it's in the setting, which means the application has erroneous configurations.

Reply:
It's just a `console` handler for  `UInt160`. If it doesn't work then it will let you know, but you'll still be able to use normal `UInt160` format strings.

Reply:
No, a configuration error should stop the application. Showing error message only if its a command. We should make the line clear here. Otherwise there will be endless argument on whether this should be an exception whether that should be a warning or error.

Reply:
It's like saying your computer OS shouldn't work, if your date and time is incorrect.

Reply:
What if they use an old `config.json`? What should happen then?

Reply:
> It's like saying your computer OS shouldn't work, if your date and time is incorrect.

HTTPS won't work :)

Reply:
> What if they use an old `config.json`? What should happen then? It has to be backwards compatible.

They will need to use the new config.json, if you use and old library with the new executable, you also will get an error, that's the same for me.

Reply:
How you figure? We use `if (section.Exists())`

Reply:
If we make this contract hash as a hard code, then nothing will be confusing. NNS hash is `[0x50ac1c37690cc2cfc594472833cf57505d5f46de` either on mainnet or testnet, it's not necessary to set it in config.json.

Reply:
@superboyiii https://github.com/neo-project/neo/pull/3032#discussion_r1431080820 @shargon wanted it there.

Reply:
@superboyiii I think its good to have in config, reason being its not a native contract and also you change it for testing.

Reply:
> @superboyiii I think its good to have in config, reason being its not a native contract and also you change it for testing.

I'm not sure if someone will test NNS contract in their privatenet.... Even though, their nns is not this nns, they can change many code, may cause your command execute failed.

Reply:
Let say for example someone run a `private net` for the node, and they want to deploy the `NNS` service to build or test a  `dapp`. If the download and compile the contract. Won't the hash be different because it's not a native contract? No sense in forcing people to use `Test net`. Especially with the gas bot or the Faucet being so slow. Say someone is trying to make a domain registrar and need to buy many many many domains for years. We can keep as is for now. If it becomes issue in future we can alway change it. You know there will be issue, like `I have a private or im using neo-express and i can't connect to NNS or it getting me error saying contract doesn't exist`.

Reply:
All the addresses that are no native contracts should be configurables